src/main/java/org/opensearch/knn/index/codec/nativeindex/BulkVectorTransferIndexBuildStrategy.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.nativeindex;
```

Comment:
> but it would be completed if we get this Strategies via Factory rather than new BulkVectorTransferIndexBuildStrategy()

The intention is to keep it singleton. A factory isn't required here since there are two cases and a simpler if else is managing it. If more cases are ever added, sure it can manage it better. For now I would prefer keeping it simple